src/internal/syscall.h

```diff
@@ -27,33 +27,104 @@ hidden long __syscall_ret(unsigned long),
 	__syscall_cp(syscall_arg_t, syscall_arg_t, syscall_arg_t, syscall_arg_t,
 	             syscall_arg_t, syscall_arg_t, syscall_arg_t);
 
-#define __syscall1(n,a) __syscall1(n,__scc(a))
```

Comment:
Can you please add in the commit message a brief description of the changes from this commit (e.g. why it's needed, what is changed). Thank you.

Also, an usage example / test scenario would be useful to be added in the PR description.